test/q-repeat.test.js

```diff
@@ -14,3 +15,56 @@ describe('map', () => {
     });
   });
 });
+
+describe('retry', () => {
+  it('repeats until success', () => {
+    let delay = 500;
+    let invocations = 0;
+
+    return Q.timeoutRetry(150, () => {
+      delay -= 100;
+      invocations += 1;
+      return Q(`answer ${invocations}`).delay(delay);
+    }).then((result) => {
+      expect(delay).to.eq(100);
+      expect(invocations).to.eq(4);
+      expect(result).to.eq('answer 4');
+    });
+  });
+
+  it('fails fast if actual error occurs', () => {
+    return Q.timeoutRetry(500, () => {
+      throw new Error('omg');
+    }).catch((err) => {
+      expect(err.message).to.eq('omg');
+    });
+  });
+
+  it('hits outer timeout', () => {
```

Comment:
I don't think that this outer timeout is a pattern you want to encourage, or use.

`Bluebird` seems to support canceling promises, but I can't find any indication that `q` does.

Without cancelling the promise somehow, the inner promise loop just keeps on spinning.
I'd suggest either just relying upon `maxRetries`, or add `maxTimeout`.

Here's an example of a run-away loop, happily spinning in the background:
https://github.com/machinemetrics/q-repeat/commit/6ff16489c3a34c32a68ba20cc0f489b8ebc8f5d8
( this branch : https://github.com/machinemetrics/q-repeat/compare/timeout/loopy )